CAP/gap/TerminalCategory.gi

```diff
@@ -4,55 +4,414 @@
+    
+    ##
+    AddIsEqualForMorphisms( T,
+      function( T, morphism1, morphism2 )
+        
+        return true;
+        
+    end );
+    
+    Finalize( T );
+    
+    return T;
+    
+end );
 
-InstallValue( CAP_INTERNAL_TERMINAL_CATEGORY,
-              
-              CreateCapCategory( "TerminalCategory" ) );
+#########################################
+#
+# Terminal category with multiple objects
+#
+#########################################
 
-SetFilterObj( CAP_INTERNAL_TERMINAL_CATEGORY, IsTerminalCategory );
+##
+InstallGlobalFunction( TerminalCategoryWithMultipleObjects,
+  function( )
+    local name, category_filter, category_object_filter, category_morphism_filter,
+          create_func_object, create_func_morphism,
+          object_constructor, object_datum, morphism_constructor, morphism_datum,
+          properties, excluded_properties, T;
+    
+    name := "TerminalCategoryWithMultipleObjects( )";
+    
+    category_filter := IsTerminalCategoryWithMultipleObjects;
+    
+    category_object_filter := IsObjectInTerminalCategoryWithMultipleObjects and HasIsZeroForObjects and IsZeroForObjects;
+    
+    category_morphism_filter := IsMorphismInTerminalCategoryWithMultipleObjects and HasIsZeroForMorphisms and IsZeroForMorphisms;
+    
+    ## e.g., ZeroObject, DirectSum
+    create_func_object :=
+        function( name, T )
+            
+            return """
+                function( input_arguments )
+                    
+                    return ObjectConstructor( cat, "operation_name" );
+                    
+                end
+            """;
+            
+        end;
+    
+    ## e.g., IdentityMorphism, PreCompose
+    create_func_morphism :=
+        function( name, T )
+            
+            return """
+                function( input_arguments )
+                    
+                    return MorphismConstructor( cat, top_source, "operation_name", top_range );
+                    
+                end
+            """;
+            
+        end;
+    
+    ##
+    object_constructor := function( cat, string )
+        
+        return ObjectifyObjectForCAPWithAttributes(
+                       rec( ), cat,
+                       String, string );
+        
+    end;
+    
+    object_datum := { cat, object } -> String( object );
+    
+    morphism_constructor := function( cat, source, string, range )
+        
+        return ObjectifyMorphismWithSourceAndRangeForCAPWithAttributes(
+                       rec( ), cat,
+                       source,
+                       range,
+                       String, string );
+        
+    end;
+    
+    morphism_datum := { cat, morphism } -> fail;
+    
+    ## prevent skeletality and strictness
+    properties := Set( List( CAP_INTERNAL_CATEGORICAL_PROPERTIES_LIST, a -> a[1] ) );
+    excluded_properties := Filtered( properties, p -> IsInt( PositionSublist( p, "Strict" ) ) );
+    Add( excluded_properties, "IsSkeletalCategory" );
```

Comment:
I want to make sure that we all are aware of the implications here:

Working with blocklists instead of allowlists always has the problem that we might not block things in the future which we actually should block. I.e., this assumes that all properties added in the future (or which we have not though about) are "probably fine", which might lead to very nasty errors if things are actually not fine. Are we prepared to live with this? @mohamed-barakat 
I would not design it in this way, but CAP does this at various places, so...^^

Reply:
Yes, I would suggest living with this for this category.